core/dbt/contracts/graph/semantic_manifest.py

```diff
@@ -59,8 +68,46 @@ def write_json_to_file(self, file_path: str):
         write_file(file_path, json)
 
     def _get_pydantic_semantic_manifest(self) -> PydanticSemanticManifest:
+        pydantic_time_spines: List[PydanticTimeSpine] = []
+        daily_time_spine: Optional[PydanticTimeSpine] = None
+        for node in self.manifest.nodes.values():
+            if not (isinstance(node, ModelNode) and node.time_spine):
```

Comment:
Why is the `and node.time_spine` bit necessary? Are there non-ModelNode nodes that have a time_spine attribute?

Reply:
Really I just want to say:
```
if not node.time_spine:
    continue
```
since the logic in this loop only applies to time spines.

The `isinstance(node, ModelNode)` is for type checking. According to mypy, node here is a `ManifestNode`, and `ModelNode` is just one option in that union. Not all of the `ManifestNode` options have the attribute `time_spine`, so I have to narrow the type here before I can access that attribute.